src/commands/executeFunction/executeEventGridFunction.ts

```diff
@@ -0,0 +1,156 @@
+        label: name
+            .replace(/\.json$/, '')
+            .split('_')
+            .map((word) => word.charAt(0).toUpperCase() + word.slice(1))
+            .join(' '),
+    }));
+    const selectedFileName: string =
+        (
+            await context.ui.showQuickPick(eventTypePicks, {
+                placeHolder: localize('selectEventType', 'Select the event type'),
+                stepName: 'eventType',
+            })
+        ).data ?? 'blob_created.json';
+
+    // Get selected contents of sample request
+    const selectedFileUrl = sampleFiles.find((fileMetadata) => fileMetadata.name === selectedFileName)?.download_url || sampleFiles[0].download_url;
+    const selectedFileContents: {} = await feedUtils.getJsonFeed(context, selectedFileUrl);
+
+    // Prompt for whether to send or modify the sample request
+    const shouldModify: boolean =
+        (
+            await context.ui.showQuickPick(
+                [
+                    {
+                        label: localize('sendSample', 'Send sample request'),
+                        data: false,
+                    },
+                    {
+                        label: localize('modifySample', 'Modify sample request'),
+                        data: true,
+                    },
+                ],
+                {
+                    placeHolder: 'Would you like to send the sample request or modify it first?',
+                    stepName: 'modifyOrSendSample',
+                },
+            )
+        ).data || false;
+
+    // Execute function with sample data directly if user chooses not to modify
+    if (!shouldModify) {
+        return executeFunctionWithInput(context, selectedFileContents, node);
+    }
+
+    // Create a temp file with the sample request & open in new window
+    const tempFilePath: string = await createTempSampleFile(eventSource, selectedFileName, selectedFileContents);
+    const document: vscode.TextDocument = await vscode.workspace.openTextDocument(tempFilePath);
+    await vscode.window.showTextDocument(document, {
+        preview: false,
+    });
+
+    // Request will be sent when the user clicks on the button or on the codelens link
+
+    // Set a listener to delete the temp file after it's closed
+    await new Promise<void>((resolve, reject) => {
+        const disposable = vscode.workspace.onDidCloseTextDocument(async (closedDocument) => {
+            if (closedDocument.fileName === document.fileName) {
+                try {
+                    await fs.unlink(tempFilePath);
+                    resolve();
+                } catch (error) {
+                    reject(error);
+                } finally {
+                    disposable.dispose();
+                }
+            }
+        });
+    });
+}
+
+async function createTempSampleFile(eventSource: string, fileName: string, contents: {}): Promise<string> {
+    const samplesDirPath = await createSamplesDirIfNotExists(eventSource);
```

Comment:
> Right now, I create the temp files that store these sample files in the .vscode directory. Would it be better if I saved it elsewhere, such as the OS temp file directory, by default?

I'd prefer to write this in the OS temp file directory personally.

It's a pretty large task, but I think long term, we should probably have a virtual file system provider similar to container apps or container apps so that we don't have to write anything in temp.

Reply:
Done